lib/chef/provider/package.rb

```diff
@@ -468,7 +476,32 @@ def package_name_array
 
       # @return [Array] candidate_version(s) as an array
       def candidate_version_array
-        [ candidate_version ].flatten
+        use_multipackage_api? ?
+          [ candidate_version ].flatten :
+          [ LazyObject.new { candidate_version } ]
+      end
```

Comment:
would it be possible to fix this so that when we're using the multipackage api here that we get back an array of lazy objects?